# Imports

In [11]:
import open3d as o3d
import numpy as np

# Running train.py

We will run the following terminal command:

```cmd
python train.py -s Gaussiatest/Test2
```

Here this command assumes there already have been input images in [`GaussianTest/Test2/input`](GaussianTest/Test2/input)

# Results

By running train.py its ouputs are stored in directory [output/Test2_Result](output/Test2_Result)

## Input.ply

In [17]:
ply_path = 'output/Test2_Result/input.ply'
pcd = o3d.io.read_point_cloud(ply_path)

print(pcd)
points = np.asarray(pcd.points)
print(points)

o3d.visualization.draw_geometries([pcd])

PointCloud with 1768 points.
[[  4.34101534  -1.97590983   8.94548512]
 [  6.54583645  10.24704552   2.88442421]
 [  4.26116467  -0.91624063   9.97420979]
 ...
 [-14.37703228  10.57718468   7.03714371]
 [ -3.41339111   4.91355276  12.40224266]
 [-14.48631001  10.56289577   6.95793295]]


### Details

Information [input.ply](output/Test2_Result/input.ply) have is the result of running convert.py. Which is 1768 point clouds.

## Results for 7000 iterations

In [15]:
ply_path = 'output/Test2_Result/point_cloud/iteration_7000/point_cloud.ply'
pcd = o3d.io.read_point_cloud(ply_path)

print(pcd)
points = np.asarray(pcd.points)
print(points)

o3d.visualization.draw_geometries([pcd])


PointCloud with 347527 points.
[[ 6.59872437 10.12901211  2.86574125]
 [ 0.22425154  1.65513897 12.2174387 ]
 [ 4.22987652 -1.58260691  9.34566116]
 ...
 [ 0.07556491  6.75840139  6.57672215]
 [ 4.2516675   7.14003801 10.84717178]
 [-0.48998436  5.14745998  7.08731127]]


## Results for 30000 iterations

In [14]:
ply_path = 'output/Test2_Result/point_cloud/iteration_30000/point_cloud.ply'
pcd = o3d.io.read_point_cloud(ply_path)

print(pcd)
points = np.asarray(pcd.points)
print(points)

o3d.visualization.draw_geometries([pcd])

PointCloud with 375184 points.
[[ 6.60873985 10.11532307  2.88246822]
 [ 0.17242943  1.56833577 12.10415363]
 [ 4.23568726 -1.58306515  9.3474226 ]
 ...
 [-1.06505299  5.10284138 12.32131481]
 [ 1.89028597  3.38411283 10.96808529]
 [-0.26036382  4.30767822 11.26516533]]
